mitodl/mit-open
# Change default ordering parameter to "sortby", add options documentation to learning_resources endpoint

# What are the relevant tickets?
Closes #220 

# Description (What does it do?)
- Changes the default DRF ordering parameter to "sortby"
- modifies the `LearningResourceFilter` class to also handle the sortby parameter and provide choices that drf_spectacular can translate to better schema documentation

# Screenshots (if appropriate):


![Screenshot 2023-11-16 at 11 12 33 AM](https://github.com/mitodl/mit-open/assets/187676/93487c97-ccb7-49eb-a5f3-3b01021cf1be)

![Screenshot 2023-11-16 at 11 13 32 AM](https://github.com/mitodl/mit-open/assets/187676/31efe0cd-e761-47b9-b233-2e230a2c8803)


![Screenshot 2023-11-16 at 11 14 42 AM](https://github.com/mitodl/mit-open/assets/187676/0f6165ba-d3e4-4cd9-8372-2322e206bd00)


# How can this be tested?

Run `./manage.py update_course_number_departments`

Run `./manage.py recreate_index --all`

Try the following urls, they should return results in the expected sort order based on the specified value:
http://localhost:8063/api/v1/learning_resources/?sortby=-id
http://localhost:8063/api/v1/learning_resources/?sortby=readable_id
http://localhost:8063/api/v1/learning_resources/?sortby=-last_modified
http://localhost:8063/api/v1/learning_resources/?sortby=start_date

Check the swagger and redoc documentation, they should look like screenshots above.
